src/reactionsystem.jl

```diff
@@ -1669,12 +1767,21 @@ function DiffEqBase.ODEProblem(rs::ReactionSystem, u0, tspan,
                                check_length = false, name = nameof(rs),
                                combinatoric_ratelaws = get_combinatoric_ratelaws(rs),
                                include_zero_odes = true, remove_conserved = false,
-                               checks = false, kwargs...)
+                               checks = false, structural_simplify = false, kwargs...)
     u0map = symmap_to_varmap(rs, u0)
     pmap = symmap_to_varmap(rs, p)
     osys = convert(ODESystem, rs; name, combinatoric_ratelaws, include_zero_odes, checks,
                    remove_conserved)
-    osys = complete(osys)
+
+    # Handles potential differential algebraic equations (which requires `structural_simplify`).
+    if structural_simplify 
+        (osys = MT.structural_simplify(osys))
+    elseif has_alg_equations(rs)
+        error("The input ReactionSystem has algebraic equations. This requires setting `structural_simplify=true` within `ODEProblem` call.")
+    else
+        osys = complete(osys)
+    end
```

Comment:
This feels like adding a keyword for something we don't own -- if a user wants to simplify why not require them to explicitly convert and call `structural_simplify`?

Reply:
But isn't it weird to add it for Problem(::ReactionSystem) but not have it for Problem(::ODESystem) (for example)? Seems like something that could confuse users? 

If you really feel strongly about it we can keep it, but I just think it introduces a strange inconsistency for the user workflow by having such a keyword just for `ReactionSystem`s.

Reply:
Generally, most users who are not developing stuff will not be dealing with non`ReactionSystem` systems anyway, I think this should be fine.